modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/disaster/ManualGroupUpdateRequest.java

```diff
@@ -0,0 +1,106 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.distributionzones.disaster;
+
+import static java.util.concurrent.CompletableFuture.allOf;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.ignite.internal.util.IgniteUtils.copyStateTo;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogZoneDescriptor;
+import org.apache.ignite.internal.distributionzones.NodeWithAttributes;
+import org.apache.ignite.internal.distributionzones.rebalance.RebalanceUtil;
+import org.apache.ignite.internal.hlc.HybridTimestamp;
+import org.apache.ignite.internal.metastorage.WatchEvent;
+import org.apache.ignite.internal.tostring.S;
+
+class ManualGroupUpdateRequest implements DisasterRecoveryRequest {
+    /** Serial version UID. */
+    private static final long serialVersionUID = 0L;
+
+    private final UUID operationId;
+
+    private final int zoneId;
+
+    private final int tableId;
+
+    ManualGroupUpdateRequest(UUID operationId, int zoneId, int tableId) {
+        this.operationId = operationId;
+        this.zoneId = zoneId;
+        this.tableId = tableId;
+    }
+
+    @Override
+    public UUID operationId() {
+        return operationId;
+    }
+
+    @Override
+    public int zoneId() {
+        return zoneId;
+    }
+
+    public int tableId() {
+        return tableId;
+    }
+
+    @Override
+    public CompletableFuture<Void> handle(
+            DisasterRecoveryManager disasterRecoveryManager,
+            WatchEvent watchEvent,
+            CompletableFuture<Void> operationFuture
+    ) {
+        long msRevision = watchEvent.revision();
+        HybridTimestamp msSafeTime = disasterRecoveryManager.metaStorageManager.timestampByRevision(msRevision);
+
+        int catalogVersion = disasterRecoveryManager.catalogManager.activeCatalogVersion(msSafeTime.longValue());
+        Catalog catalog = disasterRecoveryManager.catalogManager.catalog(catalogVersion);
+
+        CatalogZoneDescriptor zoneDescriptor = catalog.zone(zoneId);
+        CatalogTableDescriptor tableDescriptor = catalog.table(tableId);
+
+        CompletableFuture<Set<String>> dataNodesFuture = disasterRecoveryManager.dzManager.dataNodes(msRevision, catalogVersion, zoneId);
+
+        return dataNodesFuture.thenCompose(dataNodes -> {
```

Comment:
Should we set the task to another pool?

Reply:
Good question. Every other place uses it without pool. I don't do any IO here, except for sending network messages, which other usages do as well